cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1648,6 +1661,60 @@ String zkConnectionString(int connectToReplicas)  {
                     });
         }
 
+        Future<ReconciliationState> kafkaBrokerDynamicConfiguration() {
+            KafkaBrokerConfigurationHelper kbch = new KafkaBrokerConfigurationHelper(kafkaSetOperations.getAdminClientProvider(), secretOperations);
+            return kafkaSetOperations.getAsync(namespace, KafkaCluster.kafkaClusterName(name))
+                .compose(sts -> {
+                    if (sts == null) {
+                        return Future.succeededFuture();
+                    }
+                    int replicas = kafkaCluster.getReplicas();
+                    List<Future> configFutures = new ArrayList<>(replicas);
+
+                    for (int podId = 0; podId < replicas; podId++) {
+                        int finalPodId = podId;
+                        Future<Admin> acFut = kbch.adminClient(sts, podId);
+                        configFutures.add(
+                                acFut.compose(ac -> {
+                                    if (ac == null) {
+                                        kafkaPodsUpdatedDynamically.put(finalPodId, null);
```

Comment:
You set it to null here. But I didn't found any other place where you would handle the null situation. That is a bit weird, as it seems to almost as if this skips the config reconciliation for this pod:
* it doesn't update it dynamically
* it doesn't do rolling update

So should this instead fail the reconciliation?

Reply:
I think this is going to be fixed by using `podReadiness` at this place. I was hitting some situations when the pod was in `pending` state and thus it was unable to get `ac` -> skip the dynamic algorithm.

Reply:
Well, it might not be the only reason. I think the question also still stands. What is the right way to do here? Because if I understand this code right, when the client doesn't connect you have no other way to diff the configs. So you actually don't know what should or should not be done.

And because this happens for example after the fresh deployment, you cannot really fail the reconciliation either as I suggested. So I think this needs to be thought through. Maybe this has to be skipped by the initial reconciliation and fail reconciliation otherwise?

I also suspect that in reality - with the 2 seconds timeout which I saw somewhere, this will be failing quite often even in normal situations.